lib/cloud_controller/paging/pagination_options.rb

```diff
@@ -7,29 +7,49 @@ class PaginationOptions
 
     PAGE_DEFAULT      = 1
     PER_PAGE_DEFAULT  = 50
-    PER_PAGE_MAX      = 5000
     ORDER_DEFAULT     = 'id'.freeze
     DIRECTION_DEFAULT = 'asc'.freeze
-    VALID_DIRECTIONS  = %w(asc desc).freeze
 
-    attr_writer :order_by, :order_direction, :default_order_by
-    attr_accessor :page, :per_page
+    attr_writer :order_by, :order_direction, :default_order_by, :page, :per_page
+
+    validates_numericality_of :page_string,
+      allow_nil: true,
+      only_integer: true,
+      greater_than_or_equal_to: 1,
+      message: 'must be a positive integer'
+    validates_numericality_of :per_page_string,
```

Comment:
why validate on `per_page_string` rather than `per_page`?

Reply:
The validator tests a function however since we call to_i we need to validate the value of page before it gets casted to an integer.